searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp

```diff
@@ -96,11 +96,15 @@ SearchIterator::UP
 WeightedSetTermBlueprint::createLeafSearch(const fef::TermFieldMatchDataArray &tfmda, bool) const
 {
     assert(tfmda.size() == 1);
+    if ((_terms.size() == 1) && tfmda[0]->isNotNeeded() && _terms[0]->isLeaf()) {
+        return static_cast<LeafBlueprint &>(*_terms[0]).createLeafSearch(tfmda, true);
```

Comment:
need a good comment here to explain why we cannot preserve the iterator strictness

Reply:
Comment added. PTAL